### ISSUE TITLE
On user/tracks/ I think Most Commented/Fav'd/Listened Tracks aren't accurate

### ISSUE THREAD

**Issue by ofsound** [sudara/alonetone#627]
On /sudara  Suffocate tops these lists

**Comment by sudara** [comment]
The core problem here seems to be these queries seem to only be looking at a subset of the assets.

https://github.com/sudara/alonetone/blob/master/app/controllers/assets_controller.rb#L29-L32 

Perhaps only the last 30 days? Or maybe the order isn't actually correct since `@user.assets` already has an order?

<img width="527" alt="image" src="https://user-images.githubusercontent.com/472/67504394-93139f80-f689-11e9-928b-b26c62053fc3.png">

To reproduce in tests, you could create some assets with created_at over a year old, with high favorites count, high listens count, etc — and verify those top the list.

Also we should make the lists 5 tracks in length, 10 seems too long!

**Comment by innatewonderer** [comment]
Looks like `.order()` is broken:
```
2.6.3 :006 > @user.assets.most_commented.limit(10).map(&:comments_count)
  Asset Load (2.7ms)  SELECT `assets`.* FROM `assets` WHERE `assets`.`deleted_at` IS NULL AND `assets`.`user_id` = 1 AND (comments_count > 0) ORDER BY assets.id DESC, comments_count DESC LIMIT 10
 => [3, 5, 2, 2, 1, 1, 1, 1, 1, 1]
2.6.3 :007 > @user.assets.most_listened.limit(10).map(&:comments_count)
  Asset Load (1.8ms)  SELECT `assets`.* FROM `assets` WHERE `assets`.`deleted_at` IS NULL AND `assets`.`user_id` = 1 AND (listens_count > 0) ORDER BY assets.id DESC, listens_count DESC LIMIT 10
 => [3, 5, 2, 2, 1, 0, 1, 0, 0, 0]
```
Both of these methods are scopes and contain `.order('### DESC') `.

It also affects `hottest`, which at first I was wondering if `order` is not ordering floats correctly, but then found that it seems to fail to order any of them. 

I found at least one reported issue in Rails gem for version 6.0 
https://github.com/rails/rails/issues/37376
P.S. I did confirm that `limit(10)` does not impact the result.

**Comment by innatewonderer** [comment]
I noticed that they have `ORDER BY assets.id DESC, `assets`.`hotness` DESC` and I checked that it’s `@user.assets` that creates the first `ORDER BY assets.id DESC` and it appears that even if I add `order(hotness: desc)` it does not seem to have an effect on the array of records.
```
2.6.3 :004 > @user.assets.limit(10).map(&:hotness)
  Asset Load (1.9ms)  SELECT `assets`.* FROM `assets` WHERE `assets`.`deleted_at` IS NULL AND `assets`.`user_id` = 1 ORDER BY assets.id DESC LIMIT 10
 => [0.00212567, 0.0360669, 0.0426885, 0.0121963, 0.00212567, 0.00212567, 0.0288885, 0.0121963, 0.0426885, 0.0360669]
```
vs
```
2.6.3 :006 > @user.assets.hottest.limit(10).map(&:hotness)
  Asset Load (2.7ms)  SELECT `assets`.* FROM `assets` WHERE `assets`.`deleted_at` IS NULL AND `assets`.`user_id` = 1 AND (hotness > 0) ORDER BY assets.id DESC, `assets`.`hotness` DESC LIMIT 10
 => [0.00212567, 0.0360669, 0.0426885, 0.0121963, 0.00212567, 0.00212567, 0.0288885, 0.0121963, 0.0426885, 0.0360669]
```

**Comment by innatewonderer** [comment]
Fixed in #873
